integration/spark/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonCleanFilesCommand.scala

```diff
@@ -91,6 +96,14 @@ case class CarbonCleanFilesCommand(
     OperationListenerBus.getInstance.fireEvent(cleanFilesPreEvent, operationContext)
     if (tableName.isDefined) {
       Checker.validateTableExists(databaseNameOp, tableName.get, sparkSession)
+      if (forceClean) {
+        // empty the trash folder
+        TrashUtil.emptyTrash(carbonTable.getTablePath)
+      } else {
+        // clear trash based on timestamp
+        TrashUtil.deleteExpiredDataFromTrash(carbonTable.getTablePath)
+      }
+      CleanFilesUtil.cleanStaleSegments(carbonTable)
```

Comment:
this was already cleaned before moving to trash right ?

Reply:
i don't understand , what you mean here. We are cleaning the stale segments here for the first time. We delete the data in trash first based on timestamp expiration and then proceed with cleaning of stale segments and then do regular clean files operation for MFD/Compacted segments